sli/tokenarray.cc

```diff
@@ -127,13 +127,21 @@ TokenArray::toVector( std::vector< double >& a ) const
   a.reserve( size() );
   for ( Token* idx = begin(); idx != end(); ++idx )
   {
-    DoubleDatum* targetid = dynamic_cast< DoubleDatum* >( idx->datum() );
-    if ( targetid == NULL )
+    DoubleDatum* targetdd = dynamic_cast< DoubleDatum* >( idx->datum() );
+    if ( targetdd )
+    {
+      a.push_back( targetdd->get() );
+    }
+    else if ( IntegerDatum* targetid = dynamic_cast< IntegerDatum* >( idx->datum() ) )
+    {
+      a.push_back( ( double ) targetid->get() );
```

Comment:
Use the explicit cast syntax as recommended by Stroustrup:
```suggestion
      a.push_back( static_cast< double >( targetid->get() ) );
```